### PR TITLE
[depthFirstSearch] promote to proposed

### LINKED ISSUE #2201: Moving `pgr_depthFirstSearch` to proposed on 3.3
The function **pgr_depthFirstSearch - Experimental** under the traversal family of functions can be moved to proposed on version 3.3. Function's documentation can be found on https://docs.pgrouting.org/latest/en/pgr_depthFirstSearch.html

This function has all the types of tests that are required, and hence is less likely to crash the server.
- No crash tests, when the parameters are NULL.
- Types check
- Inner queries check
- Edge cases on a subset of edges & vertices of the Sample Data (for both directed and undirected graphs = 18+18)
- Another set of tests for 3-node graph, described in Issue [#1348](https://github.com/pgRouting/pgrouting/issues/1348) 
- Equivalence tests between single and multiple vertices.
- Equivalence with Breadth First Search in the form of unit tests, as they are not always equivalent.

The function belongs to a new family of function (Traversal family), and can be used to find a path from a source vertex to other vertices. If the input rows are randomly ordered, then this algorithm can be used to recommend a random path visiting all the vertices. (Refer: [Analysis of Graph input ordering for pgrouting](https://github.com/pgRouting/pgrouting/wiki/Analysis:-Graph-input-ordering))

Hence, this function can be moved to proposed on 3.3.

cc: @cvvergara

### PR DESCRIPTION
**Pull request by cvvergara**
Fixes #2201

Changes proposed in this pull request:

-   SQL files comments marked as proposed
-   Documentation files Experimental -> proposed
-   Updated NEWS and release notes

@pgRouting/admins

### CODE REVIEW

File: NEWS [hunk 1]
```diff
@@ -11,10 +11,12 @@ To see all issues & pull requests closed by this release see the [Git closed mil
 
 * [#2057](https://github.com/pgRouting/pgrouting/issues/2057): trspViaEdges columns in different order
 * [#2087](https://github.com/pgRouting/pgrouting/issues/2087): pgr_extractVertices to proposed
+* [#2201](https://github.com/pgRouting/pgrouting/issues/2201): pgr_depthFirstSearch and pgr_dijkstraNearCost to proposed
```
Comment (krashish8) [inline NEWS:14]:
```suggestion
* [#2201](https://github.com/pgRouting/pgrouting/issues/2201): pgr_depthFirstSearch to proposed
```
`pgr_dijkstraNearCost` was already moved in the last PR (same in `release_notes`)

**Review by krashish8 — APPROVED** [review]
